src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -92,6 +92,14 @@ pub trait InferCtxtExt<'tcx> {
         trait_ref: &ty::Binder<ty::TraitRef<'tcx>>,
     ) -> bool;
 
+    fn suggest_trait_object_return_type_alternatives(
```

Comment:
Btw, is there a reason we are adding a bunch of methods to this trait instead of just using free functions? It feels like a rather silly misuse of the trait system (which is intended for bounded ad-hoc polymorphism, not insisting on getting method syntax...).